packages/material-ui/src/CardHeader/CardHeader.js

```diff
@@ -1,47 +1,104 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import Typography from '../Typography';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import cardHeaderClasses, { getCardHeaderUtilityClass } from './cardHeaderClasses';
 
-export const styles = {
+const overridesResolver = (props, styles) => {
+  // const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    [`& .${cardHeaderClasses.avatar}`]: styles.avatar,
+    [`& .${cardHeaderClasses.action}`]: styles.action,
+    [`& .${cardHeaderClasses.content}`]: styles.content,
+    [`& .${cardHeaderClasses.title}`]: styles.title,
+    [`& .${cardHeaderClasses.subheader}`]: styles.subheader,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+    avatar: ['avatar'],
+    action: ['action'],
+    content: ['content'],
+    title: ['title'],
```

Comment:
Is this why we need `disableTypography` as a `styleProp`?
```suggestion
    title: [!disableTypographyt && 'title'],
```

Reply:
Don't change anything in these methods. In `styleProps` we have **all props**. The change I recommended in https://github.com/mui-org/material-ui/pull/24626#discussion_r564370246 was just so that we have the correct default. Although we are not using at this moment some of the props, we may in the future, so it's safer to include them all, so that we can avoid changes like this in the future.